server/src/main/java/org/opensearch/index/shard/CloseableRetryableRefreshListener.java

```diff
@@ -51,27 +50,47 @@ public CloseableRetryableRefreshListener(ThreadPool threadPool) {
 
     @Override
     public final void afterRefresh(boolean didRefresh) throws IOException {
-        boolean successful;
-        boolean permitAcquired = semaphore.tryAcquire();
-        try {
-            successful = permitAcquired && performAfterRefresh(didRefresh, false);
-        } finally {
-            if (permitAcquired) {
-                semaphore.release();
-            }
+        if (closed.get()) {
+            return;
         }
-        scheduleRetry(successful, didRefresh, permitAcquired);
+        runAfterRefreshWithPermit(didRefresh, () -> {});
     }
 
+    /**
+     * By default, the retry thread pool name is returned as null. The implementor has the option to override the retry
+     * thread pool name. This will be used for scheduling the retries. The method would be invoked each time when a retry
+     * is required.
+     *
+     * @return the name of the retry thread pool.
+     */
     protected String getRetryThreadPoolName() {
         return null;
     }
```

Comment:
Not in favor of keeping these values null. It is confusing since it doesn't cause the implementors to override at all and might end up picking null as the value.

Reply:
Addressed the comments.